File: day_1/PermutationsCSES.cpp
#include<iostream>
using namespace std;

// Problem Link: https://cses.fi/problemset/task/1070

int main() {
    int n; cin>>n;
    if(n==1) {
        cout<<"1";
    }
    else if(n<4) {
        cout<<"NO SOLUTION";
    }
    else {
        for(int i=2; i<=n; i+=2) cout<<i<<" ";
        for(int i=1; i<=n; i+=2) cout<<i<<" ";
    }
    return 0;
}

File: day_1/PermutationsCSES.java
import java.util.Scanner;

// Problem Link: https://cses.fi/problemset/task/1070

public class PermutationsCSES {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        if (n > 3) {
            for (int i = 2; i <= n; i += 2)
                System.out.print(i + " ");
            for (int i = 1; i <= n; i += 2)
                System.out.print(i + " ");
        } 
        else if (n == 3)
            System.out.println("NO SOLUTION");
        else
            System.out.println("1");
        sc.close();
    }
}
